sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/DeploymentController.java

```diff
@@ -52,62 +62,137 @@ AiDeploymentDeletionResponse createAndDeleteDeploymentByConfigId(
   }
 
   /**
-   * Stop all deployments with the Java specific configuration ID
+   * Create and delete a deployment with the Java specific configuration ID.
+   *
+   * @param configId The configuration id.
+   * @param accept The accept header.
+   * @return a response entity with a string representation of the deployment creation response
+   */
+  @GetMapping("/by-config/{id}/createDelete")
+  ResponseEntity<String> createAndDeleteDeploymentByConfigId(
+      @Nonnull @PathVariable("id") final String configId,
+      @RequestHeader(value = "accept", required = false) final String accept)
+      throws JsonProcessingException {
+    final var response = createAndDeleteDeploymentByConfigId(configId);
+    if ("application/json".equals(accept)) {
+      return ResponseEntity.ok()
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(mapper.writeValueAsString(response));
+    }
+    return ResponseEntity.ok("Deployment created and will be deleted.");
+  }
+
+  /**
+   * Stop all deployments with the Java specific configuration ID.
    *
    * <p>Only RUNNING deployments can be STOPPED
    *
    * @param configId The configuration id.
-   * @return the deployment modification response
+   * @param accept The accept header.
+   * @return a response entity with a string representation of the deployment modification response
    */
   @GetMapping("/by-config/{id}/stop")
   @Nonnull
-  @SuppressWarnings("unused") // debug method that doesn't need to be tested
-  List<AiDeploymentModificationResponse> stopByConfigId(
-      @Nonnull @PathVariable("id") final String configId) {
+  ResponseEntity<String> stopByConfigId(
+      @Nonnull @PathVariable("id") final String configId,
+      @RequestHeader(value = "accept", required = false) final String accept)
+      throws JsonProcessingException {
     final List<AiDeployment> myDeployments = getAllByConfigId(configId);
     log.info("Found {} deployments to STOP", myDeployments.size());
 
     // STOP my deployments
-    return myDeployments.stream()
-        .map(
-            deployment ->
-                CLIENT.modify(
-                    RESOURCE_GROUP,
-                    deployment.getId(),
-                    AiDeploymentModificationRequest.create()
-                        .targetStatus(AiDeploymentTargetStatus.STOPPED)))
-        .toList();
+    final var stoppedDeployments =
+        myDeployments.stream()
+            .map(
+                deployment ->
+                    CLIENT.modify(
+                        RESOURCE_GROUP,
+                        deployment.getId(),
+                        AiDeploymentModificationRequest.create()
+                            .targetStatus(AiDeploymentTargetStatus.STOPPED)))
+            .toList();
```

Comment:
(Minor)
Readability can be improved, I think.

```suggestion
    final var modificationRequest = AiDeploymentModificationRequest.create().targetStatus(STOPPED);
    final var stoppedDeployments =
        myDeployments.stream()
            .map(AiDeployment::getId)
            .map(id -> CLIENT.modify(RESOURCE_GROUP, id, modificationRequest))
            .toList();
```